corehq/apps/sms/tasks.py

```diff
@@ -194,11 +213,12 @@ def handle_outgoing(msg):
             # Requeue the message and try it again shortly
             return True
 
-    result = send_message_via_backend(
-        msg,
-        backend=backend,
-        orig_phone_number=orig_phone_number
-    )
+    if passes_trial_check(msg):
```

Comment:
I'm assuming if `passes_trial_check(msg)` returns false then `msg.error` will be false, and that would cause a `NameError` at

https://github.com/dimagi/commcare-hq/blob/31965b3874ff2034c46ffa0ed3c01063b78eb320/corehq/apps/sms/tasks.py#L230-L233

We may still want to consider the second scenario below even if I'm wrong about the value of `msg.error`.

I'm assuming it's intended that `bool(result) == False` when the max messages have been sent and the domain is in trial mode. Given that, what happens when `handle_unsuccessful_processing_attempt(msg)` is called in the following two scenarios:

- New domain is created in trial mode, and 1000 SMS messages are queued to be sent, but only 50 are sent due to trial mode. Later the domain is upgraded out of trial mode. Will the remaining 950 messages be sent at this point, or are they discarded?
- Domain subscription lapses. Does it go back into trial mode? Are SMS messages limited?

Reply:
if `passes_trial_check` returns `False` then we also called `set_system_error` which sets `error` to `True`.

in the first scenario you outlined, the 950 messages would be marked with an error because we called set_system_error on them, and would not be resent. they would show up in the messaging history report and on the messaging dashboard with the newly added error message.

in the second scenario, when a project is downgraded to community, there should be no way to invoke an outbound sms because all the messaging uis are invisible, all reminders and broadcasts are deactivated, and inbound sms aren't processed. so we place a hard failure if outbound sms tries to be sent without permission [here](https://github.com/dimagi/commcare-hq/blob/31965b3874ff2034c46ffa0ed3c01063b78eb320/corehq/apps/sms/api.py#L246)

Reply:
> we called set_system_error on them

I missed that.

Thanks for the detailed responses.